src/main/java/ee/tuleva/onboarding/auth/session/GenericSessionStore.java

```diff
@@ -1,49 +1,57 @@
 package ee.tuleva.onboarding.auth.session;
 
-import java.io.Serializable;
-import java.util.HashMap;
-import java.util.Map;
+import ee.tuleva.onboarding.auth.principal.AuthenticationHolder;
+import java.io.*;
 import java.util.Optional;
-import javax.servlet.http.HttpSession;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Component;
-import org.springframework.web.context.request.RequestContextHolder;
-import org.springframework.web.context.request.ServletRequestAttributes;
 
 @Component
+@RequiredArgsConstructor
 public class GenericSessionStore {
 
-  private static final String GENERIC_SESSION_STORE_ATTRIBUTES =
-      GenericSessionStore.class.getName() + ".attributes";
+  private final SessionAttributeRepository repository;
+  private final AuthenticationHolder authenticationHolder;
```

Comment:
This is not a "generic session store" anymore. It depends on the User domain, database, SessionAttributeRepository. I would rename it to just "SessionStore" or something like that

Reply:
I think the author intended "generic" as a " generic type"